lib/ActivityStreams.js

```diff
@@ -75,6 +80,27 @@ ActivityStreams.prototype = {
   },
 
   /**
+   * Every 6 hours pro-actively prime our emebedly cache with a normal content page load
+   */
+  _primeCache() {
+    setTimeout(() => {
+      PlacesProvider.links.getTopFrecentSites.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getRecentBookmarks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getRecentLinks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getFrecentLinks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      this._primeCache();
+    }, PRIME_CACHE_TIMEOUT);
```

Comment:
The function should also run at boot time. Perhaps setTimeout is called in ActivityStream's init function